### PR TITLE
DESCW-911 Dynamic / custom pattern interface

### PR DESCRIPTION
**Pull request by codewisenate**
- Adds a custom post type for "Custom Patterns"
- Registers any categorised Custom Pattern post type as patterns within the pattern picker according to the category
image
Screenshot 2023-02-01 at 1 19 48 PM
image

### CODE REVIEW

**Comment by codewisenate** [comment]
image

File: src/Setup.php [hunk 1]
```diff
@@ -415,5 +416,88 @@ public function bcgov_block_theme_wp_headers( $headers ) {
...
+			'item_scheduled'           => esc_html__( 'Custom Pattern scheduled', 'bcgov-block-theme' ),
+			'item_updated'             => esc_html__( 'Custom Pattern updated.', 'bcgov-block-theme' ),
+			'parent_item_colon'        => esc_html__( 'Parent Custom Pattern:', 'bcgov-block-theme' ),
+		];
+
+		$args = [
+			'label'                 => esc_html__( 'Custom Patterns', 'bcgov-block-theme' ),
+			'labels'                => $labels,
+			'description'           => '',
+			'public'                => true,
```
Comment (ShawnTurple) [inline src/Setup.php:472]: are we sure we want it public, can't remember if this removes it from the admin, What ever we need to do to make this cpt not accessible in the public web pages, we should do (only in admin)

File: src/Setup.php [hunk 2]
```diff
@@ -415,5 +416,88 @@ public function bcgov_block_theme_wp_headers( $headers ) {
...
+			'rest_namespace'        => 'wp/v2',
+			'has_archive'           => false,
+			'show_in_menu'          => true,
+			'show_in_nav_menus'     => true,
+			'delete_with_user'      => false,
+			'exclude_from_search'   => false,
+			'capability_type'       => 'post',
+			'map_meta_cap'          => true,
+			'hierarchical'          => false,
+			'can_export'            => false,
```
Comment (ShawnTurple) [inline src/Setup.php]: not sure i have used this one, but i would say we might want to export this.

File: src/Setup.php [hunk 3]
```diff
@@ -415,5 +416,88 @@ public function bcgov_block_theme_wp_headers( $headers ) {
...
+	 * @return void
+     */
+	public function bcgov_block_theme_register_custom_pattern() {
+
+		/**
+		 * Post Type: Custom Patterns.
+		 */
+
+		$labels = [
+			'name'                     => esc_html__( 'Custom Patterns', 'bcgov-block-theme' ),
```
Comment (ShawnTurple) [inline src/Setup.php]: can we not just use `__()` instead of `esc_html__`

File: src/Setup.php [hunk 4]
```diff
@@ -415,5 +416,88 @@ public function bcgov_block_theme_wp_headers( $headers ) {
...
+			'can_export'            => false,
+			'rewrite'               => [
+				'slug'       => 'custom-pattern',
+				'with_front' => true,
+			],
+			'query_var'             => true,
+			'menu_position'         => 20,
+			'menu_icon'             => 'dashicons-editor-kitchensink',
+			'supports'              => [ 'title', 'editor', 'revisions' ],
+			'taxonomies'            => [ 'category' ],
```
Comment (ShawnTurple) [inline src/Setup.php]: i didn't know you could do this with taxonomies, however does this share the same one with the `post` post type.  If it shares it, not sure we want to do this.

**Review by ShawnTurple — APPROVED** [review]

File: src/Setup.php (hunk 4 above)
Comment (codewisenate) [reply src/Setup.php]: Changing this to a custom taxonomy / category for Patterns.

File: src/Setup.php (hunk 3 above)
Comment (codewisenate) [reply src/Setup.php]: Yes. Done.

File: src/Setup.php (hunk 1 above)
Comment (codewisenate) [reply src/Setup.php:472]: 'publicly_queryable'    => false, <-- sets it to not accessible as a page/post. I've set this.

**Review by ShawnTurple — APPROVED** [review]

**Pull request merged**